tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/containerelement/ContainerElementConstraintSetTest.java

```diff
@@ -0,0 +1,310 @@
+		Object[] values = new Object[] { CollectionHelper.asSet( "", "First", null ) };
+
+		Set<ConstraintViolation<TypeWithSet5>> constraintViolations = getExecutableValidator().validateParameters(
+				new TypeWithSet5(),
+				method,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						)
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "c")
+	public void constructor_parameter_constraint_provided_on_type_parameter_of_a_set_gets_validated() throws Exception {
+		Constructor<TypeWithSet6> constructor = TypeWithSet6.class.getDeclaredConstructor( Set.class );
+		Object[] values = new Object[] { CollectionHelper.asSet( "", "First", null ) };
+
+		Set<ConstraintViolation<TypeWithSet6>> constraintViolations = getExecutableValidator().validateConstructorParameters(
+				constructor,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithSet6.class )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithSet6.class )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						),
+				violationWith()
+						.constraintType( NotNull.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithSet6.class )
+								.parameter( "setParameter", 0 )
+								.containerElement( "<iterable element>", true, null, null, Set.class, 0 )
+						)
+		);
+	}
+
+	private static class TypeWithSet1 {
+
+		@SuppressWarnings("unused")
+		private Set<@NotNull @NotBlank String> names;
+	}
+
+	private static class TypeWithSet2 {
+
+		@Valid
```

Comment:
Yeah, I fixed it in some of the tests but apparently missed some. Will make a global pass on all `@Valid` definitions.